model/artifact/artifact_file.go

```diff
@@ -45,6 +50,47 @@ type File struct {
 	AwsSecret string `json:"aws_secret,omitempty" bson:"aws_secret,omitempty"`
 }
 
+// stripHiddenFiles is a helper for only showing users the files they are allowed to see.
+func StripHiddenFiles(files []File, pluginUser gimlet.User) []File {
```

Comment:
I think we want to avoid having this package depend on gimlet. Instead of having the user be a param to this, I would have the second param be a boolean field for if we have a user. Then the caller can pass if we have a user